modules/ui/src/main/helper/HtmlHelper.scala

```diff
@@ -24,10 +26,16 @@ object HtmlHelper:
   def copyMeInput(content: String): Tag =
     copyMe(input(spellcheck := "false", readonly, value := content))
 
-  private def copyMe(target: Tag): Tag =
+  def copyMePgnLink(url: String, name: Frag, distinct_copy_val: Future[String]): Future[Tag] =
```

Comment:
Functions almost never take a `Future[A]` parameter. They just take a `A` and the future is handled outside the function.

Also this is the view layer and `ExecutionContext`/`Future` are strictly forbidden